plugins/chat/app/services/chat/update_thread_title_prompt.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Chat
+  # Updates the thread title prompt for a user, or if the thread
+  # does not exist, adds the user as a member of the thread before setting
+  # the thread title prompt.
+  #
+  # @example
+  # Chat::UpdateThreadTitlePrompt.call(
+  #   thread_id: 88,
+  #   channel_id: 2,
+  #   guardian: guardian,
+  #   thread_title_prompt_seen: true,
+  # )
+  #
+  class UpdateThreadTitlePrompt
+    include Service::Base
+
+    # @!method call(thread_id:, channel_id:, guardian:, thread_title_prompt_seen:)
+    #   @param [Integer] thread_id
+    #   @param [Integer] channel_id
+    #   @param [Boolean] thread_title_prompt_seen
+    #   @param [Guardian] guardian
+    #   @return [Service::Base::Context]
+
+    contract
+    model :thread, :fetch_thread
+    policy :can_view_channel
+    policy :threading_enabled_for_channel
```

Comment:
I would probably have this one first, it's a simpler call and no point doing more work if threads are disabled